File: 백준/Silver/1485. 정사각형/정사각형.py
T = int(input())

for _ in range(T):
    point = []
    for i in range(4):
        x, y = map(int, input().split())
        point.append([x, y])

    point.sort()

    #대각선
    check1 = (point[2][1] - point[1][1])**2 + (point[2][0] - point[1][0])**2
    check2 = (point[3][1] - point[0][1])**2 + (point[3][0] - point[0][0])**2
    #길이
    length1 = (point[1][1] - point[0][1])**2 + (point[1][0] - point[0][0])**2
    length2 = (point[2][1] - point[0][1])**2 + (point[2][0] - point[0][0])**2

    if check1 == check2 and length1 == length2:
        print(1)
    else:
        print(0)

File: 백준/Silver/1485. 정사각형/정사각형.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {
	public static int getSquare(int num) {
		return num * num;
	}

	public static void main(String[] args) throws IOException{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
		
		for(int i = 0; i< n; i ++) {
			int[][] point = new int[4][2];
			for(int j = 0; j < 4; j++) {
				String[] tmp = br.readLine().split(" ");
				point[j][0] = Integer.parseInt(tmp[0]);
				point[j][1] = Integer.parseInt(tmp[1]);
			}
			
			Arrays.sort(point, (p1, p2) -> {
				if(p1[0] == p2[0]) {
					return p1[1] - p2[1];
				}
				else {
					return p1[0] - p2[0];
				}
			});
			int c1 = getSquare(point[2][1] - point[1][1]) + getSquare(point[2][0] - point[1][0]);
			int c2 = getSquare(point[3][1] - point[0][1]) + getSquare(point[3][0] - point[0][0]);
			int l1 = getSquare(point[1][1] - point[0][1]) + getSquare(point[1][0] - point[0][0]);
			int l2 = getSquare(point[2][1] - point[0][1]) + getSquare(point[2][0] - point[0][0]);
			
			if(c1 == c2 && l1 ==l2) {
				System.out.println(1);
			}
			else {
				System.out.println(0);
			}
		}
	}
}